pilot/pkg/networking/core/v1alpha3/listener.go

```diff
@@ -388,10 +403,11 @@ func (configgen *ConfigGeneratorImpl) buildSidecarOutboundListeners(env model.En
 			case plugin.ListenerProtocolHTTP:
 				listenerMapKey = fmt.Sprintf("%s:%d", listenAddress, servicePort.Port)
 				if l, exists := listenerMap[listenerMapKey]; exists {
-					if !listenerTypeMap[listenerMapKey].IsHTTP() {
+					if !l.protocol.IsHTTP() {
 						conflictingOutbound.Add(1)
-						log.Warnf("buildSidecarOutboundListeners: listener conflict (%v current and new %v) on %s, destination:%s, current Listener: (%s %v)",
-							servicePort.Protocol, listenerTypeMap[listenerMapKey], listenerMapKey, clusterName, l.Name, l)
+						log.Warnf("buildSidecarOutboundListeners: listener conflict on key %s (for cluster %s), dropping listener for newer service.\n"+
+							"Current: {Protocol=%v, Created=%v}.\nNew={Protocol=%v, Created=%v}.\nListener=%v",
+							listenerMapKey, clusterName, l.protocol, l.creationTime, servicePort.Protocol, service.CreationTime, l.listener)
```

Comment:
this will confuse people.. I suggest the following (for this and below):
```
log.Warnf("buildSidecarOutboundListener: Omitting service %s due to conflict while adding to listener %s on %s. Current Protocol: %s, incoming service protocol %s", clusterName, l.listener.Name, listenerMapKey, l.protocol, servicePort.Protocol)
```

Reply:
Any Warnf we add or we have must have a matching prometheus please...

Reply:
@costinm your PR is updating prometheus ... I'm just modifying the logs in this PR to add information.